plugins/lua/sort/places.lua

```diff
@@ -111,6 +111,28 @@ local function get_location_search_key(zone)
     return table.concat(result, ' ')
 end
 
+local function get_stockpile_search_key(stockpile)
+    -- I put the stockpile_number in twice so you can search both with and without the octothorpe
+    -- (e.g. if your unnamed stockpile is listed as "Stockpile #16" you can search both "#16" or "16")
+    if #stockpile.name ~= 0 then return stockpile.name
+    else return ('Stockpile #%s %s'):format(stockpile.stockpile_number, stockpile.stockpile_number) end
+end
+
+local function get_workshop_search_key(workshop)
+    return ('%s %s %s'):format(workshop.name, df.workshop_type.attrs[workshop.type].name or '', df.workshop_type[workshop.type])
+end
+
+local function get_farmplot_search_key(farmplot)
+    local result = {}
+
+    if #farmplot.name ~= 0 then table.insert(result, farmplot.name) else table.insert(result, 'Farm Plot') end
```

Comment:
I'd think you'd want the word "farm" in the key regardless of whether the farmplot has a specific name, no?

Reply:
No, imo. This represents the default name for farms being "Farm Plot" -- there's no point in searching by 'farm' in a Farmplots page, you already know everything in that page is a farm.

Reply:
oh, of course. in my mind I was picturing the workshops page. current code looks good then